src/module-config/module-config.ts

```diff
@@ -59,6 +60,12 @@ export async function getConfig(moduleName: string): Promise<any> {
         // Recurse to config[key] and schema[key].
         const schemaPart = schema[key];
         checkForUnknownConfigProperties(schemaPart, value, keyPath + ".");
+      } else {
+        if (schema[key]["validators"]) {
+          for (let validator of schema[key]["validators"]) {
+            validator(key, value);
```

Comment:
I think the following code avoids the need for all users of this library to import and use the validator function on line 1 of validators.ts everywhere. Let me know if I'm missing something:

```ts
for (let validator of schema[key].validators) {
  const errorReasonStr = validator(key, value)
  if (errorReasonStr) {
    throw Error(`Invalid configuration value ${value} for ${keyPath}: ${errorReasonStr}`);
  }
}
```

Reply:
Yes. Fantastic observation. Thank you Joel. I've refactored in terms of this approach.